### DISCUSSION TITLE
Need help with colored icons

### DISCUSSION THREAD

**Discussion by manhtq0508** [gokcehan/lf#1794] [Q&A] [answered]
I read the documentation about [setting icons](https://github.com/gokcehan/lf/blob/master/doc.md#icons) for lf, and at the end of it, I found [a sample colored icons file](https://github.com/gokcehan/lf/blob/master/etc/icons_colored.example). I copied the entire example file and pasted it into `~/.config/lf/icons` to try it out. However, it didn’t display any icons like the image below (if I use the example file [icons.example](https://github.com/gokcehan/lf/blob/master/etc/icons.example), it works normally). I used `lf -log logs.log` to investigate what happened and found this line:

```
...
2024/09/02 22:45:18 reading icons file: expected pair but found: *.styl              00;38;2;141;193;73
...
```
Based on that log, it seems that the icons file expects only 2 columns: one for the file type and one for the icon. However, according to pull request #1674, it should be possible to use 3 columns to change the icon color. 

Some information may be helpful:
```
Distro: Arch Linux 
Kernel: 6.10.7-arch1-1 
Shell: zsh 5.9 
Terminal: kitty 0.36.1

$ lf --version
32
```

How can I correctly configure colored icons in lf?

![ksnip_20240903-005046](https://github.com/user-attachments/assets/53116f42-4217-466b-8e0d-29e8cea30a02)

**Comment by joelim-work** [comment, marked as answer]
That is because the latest released version is r32, but support for colored icons was added after that point. It is scheduled to be released in [r33](https://github.com/gokcehan/lf/milestone/1?closed=1).

There hasn't been any changes recently, and it has been close to half a year since the last release, so I think it is a good time to make this new version. This is normally handled by @gokcehan though, who I assume has been busy with other stuff lately.

**Reply by manhtq0508** [reply]
Thanks for your help
